HelpAssistHer/help-assist-her
# Creates the PregnancyCenterHistory, PregnancyCenter 'updated' object on updates

## Description

Changes:
1. Cleaned up the PregnancyCenter schema, especially `hours`
2. Generally tried to use async/await rather than callbacks or promises
3. Changed PUT to `/api/pregnancy-centers/:pregnancyCenterId` to create PregnancyCenterHistories and 'updated' objects
4. Rewrote the scripts to import data such that they use our updated schemas and use promises
5. Fixed bug in passport.deserializeUser. Changed a `find` to `findOne`

#### More info on  1. Cleaned up the PregnancyCenter schema, especially `hours`.:

PregnancyCenter hours now follow the schema: 

```
 {
    0: [{ 'open' : 900, 'close': 1500}],
    1: [],
    2: ...
    3: ...
   4: ...
   5: ...
   6: ...
}

```

0 corresponds to Sunday, and the times are in the 24-hr hhmm format as a number. (Note that because it's a number, it's not zero padded so 9am is 900 not 0900. We need it to be a number to be able to compare less than and greater than.). 

## Test Plan
yarn test

Or, to test manually:

- run `yarn data` to delete the current PregnancyCenters and import new ones
- run `yarn geocode` to geocode the new data
- do a PUT to `/api/pregnancy-centers/:pregnancyCenterId`
- Check that there is an 'updated' key on the returned PregnancyCenter object and that for each of the fields updated, there is a userId and date associated with it
- Check that there are PregnancyCenterHistory objects with PregnancyCenterIds that match the pregnancyCenterId, and which have the fields updated.
